### PR TITLE
DM-42938: Migrate content from Slack canvas

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -1,9 +1,466 @@
...
+
+1. Remove `InMemoryDatastore`. Blocked only by [RFC-965](https://jira.lsstcorp.org/browse/RFC-965).  (David) [DM-41048](https://jira.lsstcorp.org/browse/DM-41048)
+    * Have to worry about the `StorageClassDelegate` tests.
+      Move `InMemoryDatasetHandle` from `pipe_base`?
+2. Subpackage reorganization.
+   Just blocked/blocking concurrent tickets.  (Party)
+
+### Datastore records and the new consistency model (B)
+
+1. Adding records to `DatasetRef`.  [DM-40053](https://jira.lsstcorp.org/browse/DM-40053), underway.  Unblocked.  (Andy) ✅
```
Comment (andy-slac) [inline index.md]: Do we want checkmarks in the published document? I was adding them to a canvas to just track my progress.

**Review by andy-slac — APPROVED** [review]
Looks good after a quick glance 🙂 Maybe remove those checkmarks?

File: index.md [hunk 2]
```diff
@@ -1,9 +1,466 @@
...
+This builds on ideas previously considered and covered in [DMTN-242](https://dmtn-242.lsst.io) and [DMTN-249](https://dmtn-249.lsst.io).
+It also includes the initial implementation plan agreed at that meeting.
 ```
 
-## Add content here
+## Introduction
 
-See the [Documenteer documentation](https://documenteer.lsst.io/technotes/index.html) for tips on how to write and configure your new technote.
+The Rubin Data Butler was formally delivered by the construction project in 2022 {cite:p}`DMTR-271`.
+For operations the requirements were modified with the switch to a hybrid data access centre ([DMTN-240](https://dmtn-240.lsst.io); {cite:p}`DMTN-240`) where the data are served from SLAC but the users are logged into the Rubin Science Platform on Google.
```
Comment (natelust) [inline index.md]: DM guide says prefer American spelling so centre -> center

**Review by natelust — APPROVED** [review]
Was consistent with my memory of events. One minor comment

**Pull request merged**